src/codeCache.cpp

```diff
@@ -104,7 +107,19 @@ void NativeCodeCache::sort() {
     if (_max_address == NO_MAX_ADDRESS) _max_address = _blobs[_count - 1]._end;
 }
 
-const char* NativeCodeCache::binarySearch(const void* address) {
+void NativeCodeCache::setBuildId(const char *build_id, int build_id_len) {
+    _build_id = (char*)malloc(build_id_len * 2 + 1);
+    if (_build_id) {
+        char *p = _build_id;
+        for (int i = 0; i < build_id_len; i++) {
+            p += sprintf(p, "%02hhx", build_id[i]);
+        }
+        // sprintf has null-terminated it for us
+        assert(p == _build_id + build_id_len * 2);
```

Comment:
If this code is loaded into the profiled process, better not to assert 👀.

Reply:
Absolutely